internal/provider/resource_user_test.go

```diff
@@ -30,17 +31,52 @@ func TestAccPritunlUser(t *testing.T) {
 			},
 		})
 	})
+	t.Run("creates users with PIN without error", func(t *testing.T) {
+		username := "tfacc-user2"
+		orgName := "tfacc-org2"
+		pin := "123456"
+
+		check := resource.ComposeTestCheckFunc(
+			resource.TestCheckResourceAttr("pritunl_user.test", "name", username),
+			resource.TestCheckResourceAttr("pritunl_user.test", "pin", pin),
+			resource.TestCheckResourceAttr("pritunl_organization.test", "name", orgName),
+		)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { preCheck(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: testPritunlUserConfigWithPin(username, orgName, pin),
+					Check:  check,
+				},
+				// import test
+				pritunlUserImportStep("pritunl_user.test"),
+			},
+		})
+	})
 }
 
 func testPritunlUserConfig(username, orgName string) string {
-	return fmt.Sprintf(`
-		resource "pritunl_organization" "test" {
-			name = "%[2]s"
-		}
-
-		resource "pritunl_user" "test" {
-			name    			= "%[1]s"
-			organization_id		= pritunl_organization.test.id
-		}
-	`, username, orgName)
+	return testPritunlUserConfigWithPin(username, orgName, "")
+}
+
+func testPritunlUserConfigWithPin(username, orgName, pin string) string {
+	resources := fmt.Sprintf(`
+resource "pritunl_organization" "test" {
+    name = "%[2]s"
+}
+
+resource "pritunl_user" "test" {
+    name = "%[1]s"
+    organization_id = pritunl_organization.test.id
+    `, username, orgName)
+
+	if pin != "" {
```

Comment:
Let's remove this if condition and assume that the PIN is always set. it should make a template more easy and clear